### PR TITLE
Dynamic Enum Fix

### PR DESCRIPTION
**Pull request by michaeljs1990**
Adds support so that when a value that is of type Dynamic Enum is updated
after initial intake the values are cleared from the initial run and the
new ones are added.

See #546

### CODE REVIEW

**Comment by michaeljs1990** [comment]
This needs tests still but I wanted to make sure others were ok with this change before doing so.

File: app/collins/models/AssetMeta.scala [hunk 1]
```diff
@@ -201,6 +201,7 @@ object AssetMeta extends Schema with AnormAdapter[AssetMeta] with AssetMetaKeys
   }
 
   // Post enum fields, enum is not safe to extend with new values
+  type DynamicEnum = AssetMeta
```
Comment (michaeljs1990) [inline app/collins/models/AssetMeta.scala]: will remove this... please ignore.

**Comment by byxorna** [comment]
@michaeljs1990 what kind of test cases can be added to prove behavior?

**Comment by michaeljs1990** [comment]
I was going to first induct a machine and then update the asset with the same LSHW except for passing in a different value for BaseDescription, BaseProduct, BaseVendor, and BaseSerial in the xml. Currently it would show the original values from the initial induction process as well as having two entries in the database for each of the above attached to the asset.

**Comment by byxorna** [comment]
@michaeljs1990 if we can do the induction in a test, that would be ideal. That way we know nothing breaks now, and in the future with more changes that will be likely made to this codebase.

**Comment by michaeljs1990** [comment]
I think the added tests should cover all cases. Wasn't sure where to put this test though it makes use of the controller helpers a bunch but it's actually testing a sub section of AssetMeta.

File: test/collins/controllers/AssetApiHelper.scala [hunk 2]
```diff
@@ -54,9 +54,9 @@ trait AssetApiHelper extends ResponseMatchHelpers with JsonMatchers with Control
     result
   }
 
-  def updateHwInfo() = {
+  def updateHwInfo(lshwXml: String = "lshw-basic.xml") = {
     // update hw details (lshw / lldp) - cannot proceed without this
-    val lshwData = getResource("lshw-basic.xml")
+    val lshwData = getResource(lshwXml)
     val lldpData = getResource("lldpctl-two-nic.xml")
```
Comment (byxorna) [inline test/collins/controllers/AssetApiHelper.scala]: can this be parameterized, as well? While you are in there :)

Comment (michaeljs1990) [inline test/collins/controllers/AssetApiHelper.scala]: ha, for sure. I can get a second chance at travis not snuffing me on the solr test as well :laughing:

**Comment by byxorna** [comment]
Ive kicked the tests again, because i *think* `2017-05-21 07:02:08,078 - [ERROR] - SolrKey - p.a.LoggerLike$class:error:131 - Cannot create sortable multivalued key for %s, forcing non-sortable` is caused by a race.

**Comment by byxorna** [comment]
@defect @roymarantz can you take a look? This is good for landing

**Comment by michaeljs1990** [comment]
Been running this in prod for a few weeks now without issue side by side with the latest GPU changes.

**Comment by byxorna** [comment]
If we dont hear anyone reject this in a few days, I'll land this. This is an awesome fix!

**Comment by michaeljs1990** [comment]
🔥 🌶 🐝

**Pull request merged**